llvm/lib/CodeGen/VirtRegMap.cpp

```diff
@@ -532,6 +535,26 @@ bool VirtRegRewriter::subRegLiveThrough(const MachineInstr &MI,
   return false;
 }
 
+/// Check if we need to maintain liveness for undef subregister lanes that are
+/// live out of a block.
+bool VirtRegRewriter::needLiveOutUndefSubregDef(const LiveInterval &LI,
+                                                const MachineBasicBlock &MBB,
+                                                unsigned SubReg,
+                                                MCPhysReg PhysReg) const {
+  LaneBitmask UndefMask = ~TRI->getSubRegIndexLaneMask(SubReg);
```

Comment:
Assert that SubReg is not 0.

Reply:
But I think it works correctly and returns a full mask for 0?

Reply:
Probably, but then there is nothing to do here.
Up to you.